master/internal/experiment.go

```diff
@@ -484,25 +504,27 @@ func (e *experiment) isBestValidation(msg searcher.CompletedMessage) bool {
 	return isBest
 }
 
-func (e *experiment) updateState(ctx *actor.Context, state model.State) {
-	if wasPatched, err := e.Transition(state); err != nil {
+func (e *experiment) updateState(ctx *actor.Context, state model.State) (err error) {
+	var wasPatched bool
+	if wasPatched, err = e.Transition(state); err != nil {
```

Comment:
https://github.com/determined-ai/determined/pull/719#discussion_r453052624 
> so return a bool ok instead of err and ignore those in all the same places?

Reply:
Yes, you should return a bool instead. Go linting will complain if you have an unhandled error as you have now. Changing this to a bool means that you don't need all the lines above like:
`_ = e.updateState(ctx, model.StoppingErrorState)`, which can now be `e.updateState(ctx, model.StoppingErrorState)`.

More generally, you need to think about the semantics of what the error that you are returning actually is. This function used to not return an error and so the code was written under that assumption. You are now introducing an error but ignoring the error everywhere that seems important (e.g. the normal function of the program). It is confusing to understand the code if you are throwing an error that you are not gonna handle.

Reply:
if error is not returned and we set the return value to be a bool indicating whether the state was changed or not how do we handle the case that the state was changed but the database update failed? is that a successful state change or not? I assume no?

Reply:
As I mentioned in the comment below, we ignore database errors. If we fail to update the database then currently that is a successful state change. It is not ideal, but that is a much bigger change to support database error handling.